mage_ai/orchestration/monitor/monitor_stats.py

```diff
@@ -50,45 +61,106 @@ def get_stats(
         elif stats_type == MonitorStatsType.BLOCK_RUN_TIME:
             return self.get_block_run_time(**new_kwargs)
 
+    def __filter(
+        self,
+        query,
+        end_time: datetime = None,
+        pipeline_schedule_id: int = None,
+        pipeline_uuid: str = None,
+        start_time: datetime = None,
+    ) -> List:
+        filter_statement = []
+
+        if pipeline_uuid is not None:
+            filter_statement.append(PipelineRun.pipeline_uuid == pipeline_uuid)
+
+        if start_time is not None:
+            filter_statement.append(PipelineRun.created_at >= start_time)
+
+        if end_time is not None:
+            filter_statement.append(PipelineRun.created_at <= end_time)
+
+        if pipeline_schedule_id is not None:
+            filter_statement.append(PipelineRun.pipeline_schedule_id == int(pipeline_schedule_id))
+
+        if filter_statement:
+            return query.filter(*filter_statement)
+
+        return query
+
     def get_pipeline_run_count(
         self,
         pipeline_uuid: str = None,
         start_time: datetime = None,
         end_time: datetime = None,
         group_by_pipeline_type: Union[str, bool] = False,
+        pipeline_schedule_id: int = None,
         **kwargs,
     ) -> Dict:
-        pipeline_runs = self.__filter_pipeline_runs(
+        now = datetime.utcnow().timestamp()
+
+        query = (
+            PipelineRun.
+            select(
```

Comment:
@wangxiaoyou1993 @dy46 @johnson-mage we should only select the columns that are absolutely needed.